app/frontend/components/options_menu.js

```diff
@@ -115,6 +116,16 @@ class OptionsMenu {
       $(this._dictionaryOption).hide();
     }
 
+    this._localeSwitchOption = document.querySelector('#localeSwitchOption');
+    this._localeSwitchOption.addEventListener('localeChanged', async (e) => {
+      await localeController.changeLocale(e.detail.locale);
+      this.slowlyHideDisplayMenu();
+    });
+    this._localeSwitchOption.addEventListener('localeDetect', async () => {
+      await localeController.detectLocale();
+      this.slowlyHideDisplayMenu();
+    });
```

Comment:
This later can be transformed to a more "dispatcher" code style: listening to the events/actions and triggering appropriate functions to update the state